src/access/AccessControl.h

```diff
@@ -629,13 +632,15 @@ class AccessControl
 
     /**
      * Check whether access (by a subject descriptor, to a request path,
-     * requiring a privilege) should be allowed or denied.
+     * requiring a privilege) should be allowed or denied. The IM action
+     * is provided for context.
```

Comment:
It's not clear that the "IM action" is the right abstraction here, or at least it would need a lot more documentation.  For example, what happens if someone passes in MsgType::ReportData (which is the obvious action for a subscription report)?

It would be better if there were a clear "operation being performed" enum (more or less matching AccessRestrictionTypeEnum in the spec, but I think the access module has its own data types defined for things like that).  Because the fact that for events you should pass SubscribeRequest but for attributes you should pass ReadRequest even if the attribute is in fact being reported due to a subscription is (1) odd and (2) completely un-guessable from this API surface.

Plausible values for that enum would be: ReadAttribute, WriteAttribute, InvokeCommand, ReadEvent.

Also, fundamentally this value here affects how the entityId is interpreted in the RequestPath, right?  It seems like fundamentally we should be keeping that information together, in that RequestPath should have an additional member indicating what entityId really means, and we should not have a separate arg here.

Reply:
Done.

Reply:
Resolving conversation, but tracking here for post 1.4 continuation: https://github.com/project-chip/connectedhomeip/issues/34913